test/development/app-hmr/hmr.test.ts

```diff
@@ -54,7 +56,86 @@ describe(`app-dir-hmr`, () => {
       }
     })
 
+    it('should update server components after navigating to a page with a different runtime', async () => {
+      const envContent = await next.readFile(envFile)
+
+      const browser = await next.browser('/env/node')
+      await browser.loadPage(`${next.url}/env/edge`)
+      await browser.eval('window.__TEST_NO_RELOAD = true')
+
+      await next.patchFile(envFile, 'MY_DEVICE="ipad"')
```

Comment:
This is the essence of the repro:
1. goto page using Node.js runtime
2. goto page using Edge runtime (MPA)
3. Change `.env` file

What happens next varies with Webpack, Turbopack and PPR.